### DISCUSSION TITLE
Anonymous type interface implementation

### DISCUSSION THREAD

**Discussion by fuzzzeee** [dotnet/csharplang#6504] [Language Ideas] [answered]
A quick [Google](https://www.google.com/search?q=anonymous+types+interface+implement+site%3Astackoverflow.com) suggests many people have thought about this - but I don't see any open issues on the matter.

We can easily declare and consume anonymous types:

```
var array = new []{0,1,2,3};
array.Select(x => new {x});
```

Though these anonymous types can't implement an interface or abstract class. Let's say I have an interface:

```
interface MyInterface
{
	string Convert();
}
```

And I want to create an anonymous type that implements this interface, how hard would the compiler magic be to allow something similar to this syntax?

`array.Select(x=> new {x}:MyInterface{string Convert(){return x.ToString();} });`

**Comment by HaloFour** [comment, marked as answer]
See:  #738

**Reply by jnm2** [reply]
More recently, see https://github.com/dotnet/csharplang/discussions/6049
